### PR TITLE
Snackbar

### LINKED ISSUE #247: Create snackbar component/service
#### Description

We need some way to indicate success and snackbars are great for that. See https://www.figma.com/file/rBjfm2tJ2Pr9M0UNTduU9v/Joystream-Atlas?node-id=3119%3A3264

#### Requirements

- `Snackbar` component added to `src/shared/components`
  - Just the snackbar component, without logic to control it
  - Should have different variants (success/error should do)
- Some kind of service/hook allowing the use of snackbars added
  - Something like this would be great:
  ```tsx
  const displaySnackbar = useSnackbar()
  
  const onClick = () => {
    displaySnackbar({type: 'success', content: 'This is text', timeout: 3000})
  }
  ```
  - Take a look at https://github.com/iamhosseindhv/notistack, if we can use that, that's great, otherwise we should use it as an inspiration
- Component&service added to the Storybook
  - There should be a story with a snackbar permanently open, where we can see how it looks like with different props
  - There should be a story where we can use a button to open snackbars via service

### PR DESCRIPTION
**Pull request by drillprop**
Fix #247

### CODE REVIEW

File: src/hooks/useSnackbar/useSnackbar.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import { CSSTransition } from 'react-transition-group'
+
+export type DisplaySnackbarArgs = {
+  time?: number
+  variant?: 'success' | 'error' | 'info'
+  buttonText?: string
+  message: string
+}
+
+const snackbarTransitions = css`
```
Comment (mikkio-j) [inline src/hooks/useSnackbar/useSnackbar.tsx]: maybe it's better to move transition style to `Snackbar.style` to keep hook file cleaner. Or is there a reason why it's here?

File: src/shared/components/Snackbar/Snackbar.style.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import styled from '@emotion/styled'
+
+const SNACKBAR_POSITION = 34
+
+export const SnackbarWrapper = styled.div`
+  background-color: ${colors.gray[800]};
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: ${sizes(3)} ${sizes(3)} ${sizes(3)} ${sizes(6)};
```
Comment (mikkio-j) [inline src/shared/components/Snackbar/Snackbar.style.ts]:
```suggestion
  padding: ${sizes(3)};
  padding-left: ${sizes(6)};
```

File: src/shared/components/Snackbar/Snackbar.style.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { colors, sizes, typography, zIndex } from '@/shared/theme'
+import styled from '@emotion/styled'
+
+const SNACKBAR_POSITION = 34
```
Comment (mikkio-j) [inline src/shared/components/Snackbar/Snackbar.style.ts:5]: maybe we should make it less on mobile? What do you think?

File: src/hooks/useSnackbar/useSnackbar.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  displaySnackbar: (args: DisplaySnackbarArgs) => void
+  closeSnackbar: () => void
+}
+
+const SnackbarContext = createContext<SnackbarContextValue | undefined>(undefined)
+SnackbarContext.displayName = 'SnackbarContext'
+
+export const SnackbarProvider: React.FC = ({ children }) => {
+  const [isVisible, setIsVisible] = useState(false)
+  const [snackbarOpts, setsnackbarOpts] = useState<DisplaySnackbarArgs | null>(null)
```
Comment (kdembler) [inline src/hooks/useSnackbar/useSnackbar.tsx]: Should be `setSnackbarOpts`

File: src/shared/components/Snackbar/Snackbar.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+}
+
+const SnackbarComponent: React.ForwardRefRenderFunction<HTMLDivElement, SnackbarProps> = (
+  { variant = 'info', message, buttonText = 'Ok', onClick },
+  ref
+) => {
+  return (
+    <>
+      <Global styles={snackbarTransitions} />
```
Comment (kdembler) [inline src/shared/components/Snackbar/Snackbar.tsx]: Maybe it'd be a better idea to include that as part of the Snackbar context? I'm not sure what are performance implications of mounting/unmounting `Global` continuously.

File: src/shared/components/Snackbar/Snackbar.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import Button from '../Button'
+import { SnackbarProvider, useSnackbar, DisplaySnackbarArgs } from '@/hooks/useSnackbar/useSnackbar'
+
+export default {
+  title: 'Shared/Snackbar',
+  component: Snackbar,
+  args: {
+    message: 'Hello from snackbar',
+  },
+  decorators: [
```
Comment (kdembler) [inline src/shared/components/Snackbar/Snackbar.stories.tsx:13]: :+1: I had no idea that existed, will be useful

**Review by kdembler — COMMENTED** [review]
Looking good, just some minor stuff

**Review by mikkio-j — APPROVED** [review]

**Review by kdembler — APPROVED** [review]

**Pull request merged**